research_fish_analysis.py

```diff
@@ -1,288 +1,288 @@
-    """
-    Takes a two-column dataframe and plots it
-    :params: a dataframe with two columns (one labels, the other a count), a filename for the resulting chart, a title, and titles for the
-    two axes (if title is None, then nothing is plotted), and a truncate variable which cuts down the number of
-    rows plotted (unless it's 0 at which point all rows are plotted)
-    :return: Nothing, just prints a chart
-    """
-    # Set up logging
-    logger = logging.getLogger(__name__)
-    logger.info('Plotting charts...')
-    
-    if truncate > 0:
-        # This cuts the dataframe down to the number of rows given in truncate
-        dataframe = dataframe.ix[:truncate]
-
-    dataframe.plot(kind='bar', legend=None)
-    plt.title(title)
-    if xaxis != None:
-        plt.xlabel(xaxis)
-    if yaxis != None:
-        plt.ylabel(yaxis)
-    # This provides more space around the chart to make it prettier        
-    plt.tight_layout(True)
-    plt.savefig("./charts/" + filename + '.png', format = 'png', dpi = 150)
-    plt.show()
-    return
-
-
-def impact_to_txt(dataframe,colname):
-    # Set up logging
-    logger = logging.getLogger(__name__)
-    logger.info('Creating impact text file...')
-    
-    # Don't want any of the NaNs, so drop them
-    impact_dataframe = dataframe.dropna(subset=[colname])
-    # Open file for writing
-    file_for_impacts = open("./data/impact.txt", 'w')
-    # Go through dataframe row by row and write the text from the colname column to as a separate line to the text file
-    for i, row in impact_dataframe.iterrows():
-        file_for_impacts.write("%s\n" % impact_dataframe[colname][i])
-    return
-    
-    
-def check_url_status(dataframe, colname, statuscol):
-    logger = logging.getLogger(__name__)
-    logger.info('Checking URLs...')
-    
-    # Don't want any of the NaNs, so drop the rows in which NaN was entered for the URL
-    dataframe.dropna(subset=[colname], inplace=True)
-
-    h = httplib2.Http()
-    for i, row in dataframe.iterrows():    
-        try:
-            print("Checking " + dataframe[colname][i])
-            response, content = h.request(dataframe[colname][i])
-            if response.status < 400:
-                dataframe[statuscol][i] = response['date']
+#!/usr/bin/env python 
+# encoding: utf-8 
+ 
+import pandas as pd 
+from pandas import ExcelWriter 
+import numpy as np 
+import csv 
+import matplotlib.pyplot as plt 
+from urllib.parse import urlparse 
+from collections import Counter 
+import math 
+import requests 
+import httplib2 
+import logging 
+ 
+ 
+def import_xls_to_df(filename, name_of_sheet): 
+    # Set up logging 
+    logger = logging.getLogger(__name__) 
+    logger.info('Importing data...') 
+ 
+    return pd.read_excel(filename,sheetname=name_of_sheet) 
+ 
+ 
+def add_column5(dataframe,newcol): 
+    """ 
+    Adds a new column of NaNs called newcol 
+    :params: a dataframe and column name 
```

Comment:
Parameters should be individually documented. The directive is called `:param:` (singular), not `:params:` (see e.g. the [Sphinx RTD Tutorial](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html)). For example:
```suggestion
:param dataframe: the input dataframe
:param newcol: name of the new NaN column to be added
```

Reply:
By the way, for readability it may be nice to switch to the [Google or NumPy style](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#google-vs-numpy) for docstrings. Both of these are much easier to parse for the reader than reStructured Text style docstrings.